### ISSUE TITLE
How does logging work?

### ISSUE THREAD

**Issue by sjmittal** [ContainerSolutions/java-operator-sdk#101]
Hi,
Sorry I am new to Kubernetes and the operator sdk itself.
I was trying out few samples.
Like in `webserver` sample we have a console logger configured.
However when I deploy the webserver the operator kicks in and does the job, but I don't see any console logs.
How can I view the same?
Also if I configure the logger to log to a file, where would the file be generated?

I am trying a local deployment using minikube on windows.

Any help would be greatly appreciated in trying to understand how logging works, so as to better understand the operator framework.

Thanks
Sachin

**Comment by JRial95** [comment]
You can view the logs using :
```
kubectl logs <pod-name>
```
More information about this command [here](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#logs).

If you configure the logger to log to a file, the file will be generated inside the container itself. Keep in mind that a pod is a relatively ephemeral entity and can be replaced at any time by an identical pod on another node. In this case, you will lose your log file.

**Comment by sjmittal** [comment]
Hi,
I was able to dump the logs and also found other commands to get cluster info, node info, pod info etc. I am able to deploy the webserver operator and niginx webserver and expose it at service. However I am not able to connect to that service. From the logs I don't see any errors, so wanted to understand how can I debug this issue better.
 Here are some relevant details of commands I executed:
I am running a single node local cluster using minikube for windows.
- So once I get the node details
`kubectl describe node minikube`
Response (I also get the list of pods):
```
Addresses:
  InternalIP:  172.17.0.2
  Hostname:    minikube
```
- and when I get the helllows pod details
`kubectl describe pod hellows-9467f7b4f-zm4tx`
Response
```
Node:           minikube/172.17.0.2
Status:         Running
IP:             172.18.0.7
```
- also when I try to get service details and urls (via `kubectl`)
`kubectl describe service hellows`
Resoonse:
```
Type:                     NodePort
IP:                       10.105.56.56
Port:                     <unset>  80/TCP
TargetPort:               80/TCP
NodePort:                 <unset>  30226/TCP
Endpoints:                172.18.0.7:80
```
-  and via `minikube` commands
`minikube service -n webserver-operator --url hellows`
Response
`http://172.17.0.2:30226`
- Finally I try to access the nginx server via browser:
`http://172.18.0.7:80` or `http://172.17.0.2:30226`
none works.
_Also note that ping to these ip address also does not return anything._

I am unable to understand if my nginx server is running as described by the pods here:
```
Containers:
  nginx:
    Container ID:   docker://8e13e563ac62979391ef8e3273e89320d65ca7937171e34e3888f3276a531b60
    Image:          nginx:1.17.0
    Image ID:       docker-pullable://nginx@sha256:bdbf36b7f1f77ffe7bd2a32e59235dff6ecf131e3b6b5b96061c652f30685f3a
    Port:           80/TCP
    Host Port:      0/TCP
    State:          Running
      Started:      Sat, 09 May 2020 10:47:40 +0530
    Last State:     Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Fri, 08 May 2020 23:31:16 +0530
      Finished:     Sat, 09 May 2020 00:58:15 +0530
    Ready:          True
```
why I am unable to access it from outside.
Is there some issue with the nginx deployment via webserver operator or some issue with my kubernetes or docker configuration itself.

How can I debug this further.
Thanks
Scahin

**Comment by sjmittal** [comment]
I think there is a fix for this issue (for windows) discussed in:
https://github.com/kubernetes/minikube/issues/7644
It would be available in minikube version `minikube v1.10.0`

I guess for now a port forward command of `kubectl` works.
`kubectl port-forward pod/<pod name> <localhost port>:<target port>`    
eg:
`kubectl port-forward pod/hellows-9467f7b4f-zm4tx 30226:80`

`http://localhost:30226/` works!!!

or doing 
minikube ssh
and requesting the resource via `curl` using:
`<node ip>:<node port>`
or
`<pod ip>:<target port>`
works!!!

**Comment by csviri** [comment]
@sjmittal great, we also create a service inside the controller, it's  quite easy to expose it from that point, pls see https://kubernetes.io/docs/tutorials/hello-minikube/#create-a-service

@adam-sandor we probably need to improve the docs regarding to this in the example.
